### PR TITLE
link openssl statically for Windows

### PR DESCRIPTION
**Pull request by yihau**
#### Problem

a refined version of the timeline for future reference:

- we applied the `vendored` feature on openssl for all platforms.
- it looks like we had some perl issues so we did `Unvendor OpenSSL for Windows to avoid CI troubles with perl.exe`https://github.com/anza-xyz/agave/commit/c9bfc99c720ed091c326c1cce58fcd8197306235
- we planed to vendor Windows binaries #73.
- the pipeline failed recently https://github.com/anza-xyz/agave/actions/runs/11163589550/job/31030913975 and I have this PR

#### Summary of Changes

- use `vendored` in openssl for all platforms (it should work. I verified by this pipeline https://github.com/yihau/solana/actions/runs/11241556386/job/31253379456#step:7:8654)
```
> ldd solana-test-validator
  	ntdll.dll => /c/Windows/SYSTEM32/ntdll.dll (0x7ffe8ccc0000)
  	KERNEL32.DLL => /c/Windows/System32/KERNEL32.DLL (0x7ffe8b580000)
  	KERNELBASE.dll => /c/Windows/System32/KERNELBASE.dll (0x7ffe8a240000)
  	apphelp.dll => /c/Windows/SYSTEM32/apphelp.dll (0x7ffe88280000)
  	bcryptprimitives.dll => /c/Windows/System32/bcryptprimitives.dll (0x7ffe8ab70000)
  	USER32.dll => /c/Windows/System32/USER32.dll (0x7ffe8b3d0000)
  	win32u.dll => /c/Windows/System32/win32u.dll (0x7ffe8ab40000)
  	POWRPROF.dll => /c/Windows/SYSTEM32/POWRPROF.dll (0x7ffe8a120000)
  	GDI32.dll => /c/Windows/System32/GDI32.dll (0x7ffe8c610000)
  	ucrtbase.dll => /c/Windows/System32/ucrtbase.dll (0x7ffe8a800000)
  	gdi32full.dll => /c/Windows/System32/gdi32full.dll (0x7ffe8a6e0000)
  	RPCRT4.dll => /c/Windows/System32/RPCRT4.dll (0x7ffe8c1f0000)
  	msvcp_win.dll => /c/Windows/System32/msvcp_win.dll (0x7ffe8aaa0000)
  	CRYPT32.dll => /c/Windows/System32/CRYPT32.dll (0x7ffe8a910000)
  	WS2_32.dll => /c/Windows/System32/WS2_32.dll (0x7ffe8b7d0000)
  	ADVAPI32.dll => /c/Windows/System32/ADVAPI32.dll (0x7ffe8b710000)
  	msvcrt.dll => /c/Windows/System32/msvcrt.dll (0x7ffe8b320000)
  	sechost.dll => /c/Windows/System32/sechost.dll (0x7ffe8c640000)
  	bcrypt.dll => /c/Windows/System32/bcrypt.dll (0x7ffe8aa70000)
  	SHLWAPI.dll => /c/Windows/System32/SHLWAPI.dll (0x7ffe8c5b0000)
  	shell32.dll => /c/Windows/System32/shell32.dll (0x7ffe8b8d0000)
  	secur32.dll => /c/Windows/SYSTEM32/secur32.dll (0x7ffe7db70000)
  	ole32.dll => /c/Windows/System32/ole32.dll (0x7ffe8b1e0000)
  	combase.dll => /c/Windows/System32/combase.dll (0x7ffe8c910000)
  	VCRUNTIME140.dll => /c/Windows/SYSTEM32/VCRUNTIME140.dll (0x7ffe7e9a0000)
  	MSVCP140.dll => /c/Windows/SYSTEM32/MSVCP140.dll (0x7ffe7e9c0000)
  	VCRUNTIME140_1.dll => /c/Windows/SYSTEM32/VCRUNTIME140_1.dll (0x7ffe7f450000)
  	CRYPTBASE.DLL => /c/Windows/SYSTEM32/CRYPTBASE.DLL (0x7ffe899b0000)
  	SSPICLI.DLL => /c/Windows/SYSTEM32/SSPICLI.DLL (0x7ffe89bf0000)
```
- fixed openssl version to 3.3.2
- update ci

### CODE REVIEW

**Comment by yihau** [comment]
(will re-enable those 2 pipelines when this one is ready to merge)

File: .github/workflows/cargo.yml [hunk 1]
```diff
@@ -41,11 +41,15 @@ jobs:
         with:
           version: "v0.8.1"
 
-      - shell: bash
-        run: .github/scripts/cargo-clippy-before-script.sh ${{ runner.os }}
+      - name: Set Perl environment variables
+        if: runner.os == 'Windows'
+        run: |
+          echo "PERL=$((where.exe perl)[0])" | Out-File -FilePath $env:GITHUB_ENV -Append -Encoding utf8
+          echo "OPENSSL_SRC_PERL=$((where.exe perl)[0])" | Out-File -FilePath $env:GITHUB_ENV -Append -Encoding utf8
```
Comment (joncinque) [inline .github/workflows/cargo.yml:49]: Could you add a comment linking to this issue https://github.com/sfackler/rust-openssl/issues/2149? I was very confused about this line, and I'm assuming this helped you too 😄

File: .github/workflows/release-artifacts.yml [hunk 2]
```diff
@@ -39,14 +39,18 @@ jobs:
         run: |
           git checkout ${{ inputs.commit }}
 
+      - name: Set Perl environment variables
+        run: |
+          echo "PERL=$((where.exe perl)[0])" | Out-File -FilePath $env:GITHUB_ENV -Append -Encoding utf8
+          echo "OPENSSL_SRC_PERL=$((where.exe perl)[0])" | Out-File -FilePath $env:GITHUB_ENV -Append -Encoding utf8
```
Comment (joncinque) [inline .github/workflows/release-artifacts.yml:46]: Same here, can you add a comment linking to this issue? https://github.com/sfackler/rust-openssl/issues/2149

**Review by joncinque — COMMENTED** [review]
Apologies for the lateness! The bin in the release works for me on windows, even in cmd.exe, and I was able to (eventually) build this on my local Windows box, so the change looks good! I got a bit confused about the Perl stuff, so just a nit to add a comment

File: .github/workflows/cargo.yml (hunk 1 above)
Comment (yihau) [reply .github/workflows/cargo.yml:49]: sure! [84af4f9](https://github.com/anza-xyz/agave/pull/3118/commits/84af4f95f94f2e0afdd220075eb34dd94d795cc3) 🫶

File: .github/workflows/release-artifacts.yml (hunk 2 above)
Comment (yihau) [reply .github/workflows/release-artifacts.yml:46]: put it in the same commit! [84af4f9](https://github.com/anza-xyz/agave/pull/3118/commits/84af4f95f94f2e0afdd220075eb34dd94d795cc3)

**Review by joncinque — APPROVED** [review]
Thanks for taking care of this!

**Pull request merged**